docs/cpp/faq-cpp.md

```diff
@@ -0,0 +1,79 @@
+---
+Order: 15
+Area: cpp
+TOCTitle: FAQ
+ContentId: EC8DC085-A0E4-4401-B41F-6497EDD49352
+PageTitle: C/C++ extension FAQ
+DateApproved: 06/25/2019
+MetaDescription: Frequently asked questions about the C/C++ extension in Visual Studio Code.
+---
+# Frequently asked questions
+
+- [Why are my files corrupted on format?](#why-are-my-files-corrupted-on-format)
+- [How do I get IntelliSense to work correctly?](#how-do-i-get-intellisense-to-work-correctly)
+- [Why do I see red squiggles under Standard Library types?](#why-do-i-see-red-squiggles-under-standard-library-types)
+- [How do I get the new IntelliSense to work with MinGW on Windows?](#how-do-i-get-the-new-intellisense-to-work-with-mingw-on-windows)
+- [How do I get the new IntelliSense to work with the Windows Subsystem for Linux?](#how-do-i-get-the-new-intellisense-to-work-with-the-windows-subsystem-for-linux)
+- [What is the difference between `includePath` and `browse.path` in **c_cpp_properties.json**?](#what-is-the-difference-between-includepath-and-browsepath-in-c_cpp_propertiesjson)
+- [How do I re-create the IntelliSense database?](#how-do-i-re-create-the-intellisense-database)
+
+## Why are my files corrupted on format?
+
+Files can be corrupted due to the fact that you either have a multi-root workspace where one folder is a child of the other, or you are using symlinks to open your file. Reduce the folders in the workspace to one and remove the symlink. This should fix your problem.
+
+## How do I get IntelliSense to work correctly?
+
+There are two IntelliSense engines present in the extension: the "fuzzy" engine (or Tag Parser), and the new "Default" engine. If you are using version 0.11.0 or higher of the cpptools extension, then you can preview the new IntelliSense engine, which has more accurate autocomplete suggestions and tooltips. To use the new engine, you need to ensure that `C_Cpp.intelliSenseEngine` is set to `Default` in your settings. Since the engine is still in preview, it is not on by default for everyone yet.
```

Comment:
This FAQ needs an update.

Reply:
I need some specifics.

Reply:
Here's the replacement markdown (remove the two sub-sections below this one too):
```
Without any configuration, the extension will attempt to locate headers by searching your workspace folder and by emulating a compiler it finds on your computer. (e.g. cl.exe/WSL/MinGW for Windows, gcc/clang for macOS/Linux). If this automatic configuration is insufficient, you can modify the defaults by running the `C/C++: Edit Configurations (UI)` command. In that view, you can change the compiler you wish to emulate, the paths to include files you wish to use, preprocessor definitions, and more.

Or, if you install a build system extension that interfaces with our extension, you can allow that extension to provide the configurations for you. For example, the CMake Tools extension can configure projects that use the CMake build system. Use the `C/C++: Change Configuration Provider...` command to enable any such extension to provide the configurations for IntelliSense.

A third option for projects without build system extension support is to use a [`compile_commands.json`](https://clang.llvm.org/docs/JSONCompilationDatabase.html) file if your build system supports generating this file. In the "Advanced" section of the Configuration UI, you can supply the path to your `compile_commands.json` and the extension will use the compilation information listed in that file to configure IntelliSense.

> **Note:** If the extension is unable to resolve any of the `#include` directives in your source code, it will not show linting information for the body of the source file. If you check the Problems window in VS Code, the extension will provide more information about which files it was unable to locate. If you want to show the linting information anyway, you can change the value of the `C_Cpp.errorSquiggles` setting.
```